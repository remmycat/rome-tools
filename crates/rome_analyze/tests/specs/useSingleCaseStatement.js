switch (foo) {
    case true:
    case false:
        let foo = '';
        foo;
}

switch (foo) {
    case true:
    case false:
        'yes';
}

switch (foo) {
    case true: {
        // empty
    }
}

switch (foo) {
    case true:
}
