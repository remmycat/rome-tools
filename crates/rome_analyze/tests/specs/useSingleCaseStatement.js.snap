---
source: crates/rome_analyze/tests/spec_tests.rs
expression: useSingleCaseStatement.js
---
# Input
```js
switch (foo) {
    case true:
    case false:
        let foo = '';
        foo;
}

switch (foo) {
    case true:
    case false:
        'yes';
}

switch (foo) {
    case true: {
        // empty
    }
}

switch (foo) {
    case true:
}

```

# Diagnostics
```
error[useSingleCaseStatement]: A switch case should only have a single statement.
  ┌─ useSingleCaseStatement.js:3:5
  │  
3 │ ┌     case false:
4 │ │         let foo = '';
5 │ │         foo;
  │ └────────────^ A switch case should only have a single statement.

Wrap the statements in a block
    | @@ -1,8 +1,9 @@
0 0 |   switch (foo) {
1 1 |       case true:
2   | -     case false:
  2 | +     case false: {
3 3 |           let foo = '';
4 4 |           foo;
  5 | +     }
5 6 |   }
6 7 |   
7 8 |   switch (foo) {


```


